tests/integration/multicluster/multinetwork/multinetwork_test.go

```diff
@@ -0,0 +1,158 @@
+
+package multinetwork
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	primary istio.Instance
+	remote  istio.Instance
+	// env     *kube.Environment
+)
+
+// This test requires `--istio.test.env=kube` and at least two clusters on --istio.test.kube.config
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("multicluster_multi_network_integration_test", m).
+		RequireEnvironment(environment.Kube).
+		Setup(func(ctx resource.Context) (err error) {
+			env := ctx.Environment().(*kube.Environment)
+			if len(env.KubeClusters) >= 2 {
+				return nil
+			}
+			return fmt.Errorf("this test requires 2 KubeConfigs, like  --istio.test.kube.config config-a:config-b")
+		}).
+
+		// Deploy Istio
+		SetupOnEnv(environment.Kube, istio.Setup(&primary, nil)).
+		SetupOnEnv(environment.Kube, istio.Setup(&remote, setupConfigIndex1)).
+		Run()
+}
+
+func setupConfigIndex1(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	// We want to setup Istio on the 2nd supplied KubeConfig m(index of 1)
+	cfg.KubeIndex = 1
+}
+
+// Do a test of istioctl against first cluster
+func TestIstioCtlAgainstFirstKubeConfig(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			g := galley.NewOrFail(t, ctx, galley.Config{})
+			_ = pilot.NewOrFail(t, ctx, pilot.Config{Galley: g})
+
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			args := []string{"version", "--remote=true"}
+
+			output, fErr := istioCtl.Invoke(args)
+
+			if fErr != nil {
+				t.Fatalf("Unwanted exception for 'istioctl %s': %v", strings.Join(args, " "), fErr)
+			}
+
+			// istioctl will return a single "control plane version" if all control plane versions match
+			controlPlaneRegex := regexp.MustCompile(`control plane version: [a-z0-9\-]*`)
+			if controlPlaneRegex.MatchString(output) {
+				return
+			}
+
+			t.Logf("Did not find control plane version. This may mean components have different versions.")
+
+			// At this point, we expect the version for each component
+			expectedRegexps := []*regexp.Regexp{
+				regexp.MustCompile(`citadel version: [a-z0-9\-]*`),
+				regexp.MustCompile(`client version: [a-z0-9\-]*`),
+				regexp.MustCompile(`egressgateway version: [a-z0-9\-]*`),
+				regexp.MustCompile(`ingressgateway version: [a-z0-9\-]*`),
+				regexp.MustCompile(`pilot version: [a-z0-9\-]*`),
+				regexp.MustCompile(`galley version: [a-z0-9\-]*`),
+				regexp.MustCompile(`policy version: [a-z0-9\-]*`),
+				regexp.MustCompile(`sidecar-injector version: [a-z0-9\-]*`),
```

Comment:
I hope this fails tests.. this shouldn't exist anymore

Reply:
And I am not sure this is the right test here, tying tests to component names caused us a bit of pain in the past

Reply:
It also doesn't really seem to test any multicluster aspects. I realize this is probably just a "hello world" though

Reply:
Yeah, didn't realize this was part of the PR. Removed and will replace with a more appropriate multi-cluster test in a follow-on.